### DISCUSSION TITLE
[Feature] Allow setting custom unique user ids

### DISCUSSION THREAD

**Discussion by klejejs** [immich-app/immich#2393] [Feature Request]
### The feature

As the photos for each user are stored in their respective user id folder, it would be nice if custom user ids could be set to make storage folders more readable when opened with other file managers.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by jrasm91** [comment]
Maybe we can instead add a separate (unique) field for the user which could be a custom identifier for the storage path. As long as it is valid to use in the path and unique it should be fine. It is probably best to keep using uuids for user id field though.

**Reply by bo0tzz** [reply]
Adding a username field and allowing that in the template seems like a good idea.
